### DISCUSSION TITLE
Command not working as expected

### DISCUSSION THREAD

**Discussion by blaketeres** [community/community#26188] [Actions]
<p>I am trying to install and use the <code>ag</code> command for my action. I am able to successfully install it using <code>sudo apt-get</code>. However, it seems to behave erratically. I have tried many combinations of invoking the command, but to no luck. It always errors out with</p>
<pre><code>##[error]Process completed with exit code 1.
</code></pre>
<p>At the bottom of this post is a minimal working example yaml block. The command does not error out when running</p>
<pre><code>ag --help
</code></pre>
<p> in the action but does not work with any other combination of arguments. I do not experience any errors locally on my machine, so I do not think it is misuse of the command.</p>
<p>Am I doing something wrong? I’ve spent a day and a half on this now, and no luck. I have no idea why this isn’t working, as the built in linux commands work fine. Apologies if the answer is simple.</p>
<p>Thanks in advance.</p>
<pre><code>name: test

on:
  push:
    branches:
      - master

jobs:
  test-job:
    runs-on: ubuntu-latest

    steps:

    - name: Install and Run Silver Searcher
      run: |
        sudo apt-get install silversearcher-ag
        ag --no-color --group "some_search_string"
</code></pre>

**Comment by weide-zhou** [comment]
<p>Thank you for reaching this out!</p>
<p>I can reproduce the same with ‘ag’ command. I have raised an internal ticket to github. I will update once there’s an update. </p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/blaketeres">@blaketeres</a> ,</p>
<p>It appears that ag will return 0 if it can find the string and 1 if it can’t.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7df86c69ce19976bb8e393b82de0b1dd7c149f5a.png" data-download-href="/uploads/short-url/hYnVzPSm7Rv47gUpgWCtymkdhZU.png?dl=1" title="errorcode.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181095993-7fd7c32e-5f1a-4284-b2d9-bbb8c6d8c93d.png" alt="errorcode.png" data-base62-sha1="hYnVzPSm7Rv47gUpgWCtymkdhZU" width="690" height="257" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7df86c69ce19976bb8e393b82de0b1dd7c149f5a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">errorcode.png</span><span class="informations">736×275 36.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>A nonzero exit code will cause the step to fail. You can set ‘continue-on-error: true’ in the step if this behavior isn’t desired: <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error</a></p>
<p>Thanks.</p>

**Comment by blaketeres** [comment]
<p>Yes, that makes sense. However, <code>ag</code> is returning 1 when my search string actually exists. I apologize for not being more clear and providing a better example. See the example below.</p>
<pre><code>name: test

on:
  push:
    branches:
      - master

jobs:
  test-job:
    runs-on: ubuntu-latest

    steps:
    - name: Install Silver Searcher
      run: |
        sudo apt-get install silversearcher-ag

    - name: Make test search file
      run: echo "hello world" &gt; test_search.txt

    - name: Run Silver Searcher on howto's
      run: |
        ag "hello" # this should not fail ("hello" found inside of "test_search.txt:)
</code></pre>
<p>This <code>ag</code> command should not return 1, but should return 0. I am experiencing a failing <code>ag</code> command with a return value of 1 in this case. Am I managing the paths in the container incorrectly?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/blaketeres">@blaketeres</a> ,</p>
<p>You can add ‘exit 0’ at the end of the ag command, it works on my side. Code sample as below:</p>
<pre><code>steps:
      - name: Install Silver Searcher
        run: |
          sudo apt-get install silversearcher-ag
      - name: Make test search file
        run: echo "hello world" &gt; test_search.txt

      - name: Run Silver Searcher on howto's
        run: |
          ag "hello" &amp; exit 0
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181095995-695168f3-49da-475e-8105-c9ca0b82c4c3.png" alt="agcommand.png" data-base62-sha1="d3TLozVVHDLBfN7DpbAEiD34O2d" width="335" height="265"></p>
<p>Thanks.</p>

**Reply by fabn** [reply]
I spent hours debugging this and still not clear to me why `& exit 0` is needed since in this case `ag` finds a result so its exit code is 0, a simple `run: ag hello` should work in CI however is not working as intended
